marimo/_runtime/packages/pypi_package_manager.py

```diff
@@ -378,6 +378,26 @@ async def uninstall(self, package: str) -> bool:
         )
 
     def list_packages(self) -> list[PackageDescription]:
+        # First try with `uv tree`
+        tree = self.dependency_tree()
+        if tree is not None:
+            LOGGER.info("Listing packages with 'uv tree'")
+            seen: set[str] = set()
```

Comment:
```python
seen: dict[Name, set[Version]] = {}
...

if pkg.name in seen:
    seen[pkg.name].add({pkg.version})
else:
    seen[pkg.name] = set({pkg.version})
...

packages = [PackageDescription(name, ",".join(sorted(versions))) for pkg, versions in seen.items()]
```

Are there many dupes? We likely want to report them all rights?

Reply:
there are dups, but that is expected since we flatten the tree

Reply:
There can only be one version of a Python dependency when uv solves an environment, it will fail otherwise. Thus, a dependency can be found in multiple parts of the tree, but will have the same version, so this code seems correct to me.